ledger-core/testutils/journal/journal.go

```diff
@@ -102,10 +103,10 @@ func (p *Journal) WaitStopOf(sample smachine.StateMachine, stopCount int) syncki
 		).AfterPositiveToZero)
 }
 
-func (p *Journal) WaitAnyActivityOf(sample smachine.StateMachine, stopCount int) synckit.SignalChannel {
+func (p *Journal) WaitFutureSlotInited(stopCount int) synckit.SignalChannel {
 	return p.WaitOnce(
 		predicate.NewCounter(
-			predicate.NewSMTypeFilter(sample, nil),
+			predicate.NewSMTypeFilter(&conveyor.FutureEventSM{}, predicate.BeforeStep((&conveyor.FutureEventSM{}).StepWaitMigration)),
```

Comment:
as mentioned earlier - you are mixing different cases

and Journal is NOT a right place for this specific predicate as well